qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java

```diff
@@ -0,0 +1,241 @@
+	public void userClicksOnSaveAppointmentRequest() {
+		dashboardPage = requestAppointmentPage.saveRequest();
+	}
+
+	@Then("the system adds appointment request into the appointment requests table")
+	public void systemSavesAppointmentRequest() {
+		assertNotNull(dashboardPage.getAppointmentRequestsList());
+	}
+
+	@When("a user clicks on Schedule Appointment link from the Patient dashboard")
+	public void loadManageAppointmentsPage() {
+		manageAppointmentsPage = (ManageAppointmentsPage) dashboardPage.goToManageAppointments();
+	}
+
+	@And("the system loads Manage Appointments page")
+	public void systemLoadsManageAppointmentsPage() {
+		assertTrue(manageAppointmentsPage.containsText(testPatient.identifier));
+	}
+
+	@And("a user clicks on Cancel this Request icon")
+	public void userClicksOnDeleteAppointmentRequest() {
+		manageAppointmentsPage.deleteRequest() ;
+	}
+
+	@Then("the system deletes appointment request from the request table")
+	public void systemDeletesAppointmentRequest() {
+		assertTrue(manageAppointmentsPage.containsText("No appointment requests"));
+	}
+
+	@When("a user clicks Appointment scheduling app from the home page")
+	public void loadAppointmentSchedulingPage() {
+		dashboardPage.goToHomePage();
+		appointmentSchedulingPage = (AppointmentSchedulingPage ) homePage.goToAppointmentScheduling();
+	}
+
+	@Then("the system loads the appointment scheduling page")
+	public void systemLoadsAppointmentSchedulingPage() {
+		assertTrue(textExists("Appointment Scheduling"));
+	}
+
+	@And("a user clicks on Manage Provider Schedules app")
+	public void loadManageProviderSchedulesPage() {
+		manageProviderSchedulesPage = (ManageProviderSchedulesPage) appointmentSchedulingPage.goToManageProviderSchedules();
+	}
+
+	@And("the system loads Manage Provider Schedules page")
+	public void systemLoadsManageProviderSchedulesPage() {
+		assertTrue(textExists("Manage Appointment Blocks"));
+	}
+
+	@And("a user fills in the details of the appointment block")
+	public void captureAppointmentBlockDetails() {
+		manageProviderSchedulesPage.selectLocation(LOCATION);
+		manageProviderSchedulesPage.clickOnNextWeekday();
+		manageProviderSchedulesPage.selectLocationBlock(LOCATION);
+		manageProviderSchedulesPage.enterMinimumTimeValue("06", "30");
+		manageProviderSchedulesPage.clickOnStartTimeButton();
+		manageProviderSchedulesPage.enterMaximumTimeValue("09", "30");
+		manageProviderSchedulesPage.clickOnEndTimeButton();
+		manageProviderSchedulesPage.enterService(FIRST_SERVICE_NAME);
+	}
+
+	@And("a user clicks on the save button")
+	public void saveAppointmentBlockDetails() {
+		manageProviderSchedulesPage.clickOnSave();
+	}
+
+	@Then("the system saves the appointment block")
+	public void systemSavesAppointmentBlockDetails() {
+		assertTrue(manageProviderSchedulesPage.containsText(FIRST_SERVICE_NAME));
+		dashboardPage.goToHomePage();
+	}
+
+	@And("a user clicks on the Manage Provider Schedules app")
+	public void loadAppointmentBlocksPage() {
```

Comment:
The gherkin is saying user click on manage provider schedules app but the functionality is showing go to appointment block. anything to clarify here

Reply:
> The gherkin is saying user click on manage provider schedules app but the functionality is showing go to appointment block. anything to clarify here

@sherrif10 when the user clicks on manage provider schedules app , it is the appointment blocks page that is loaded. Also I realized that depending on the functionalities that you want, it can either be the appointment blocks page or the manage providers schedules  page that loads.